### PR TITLE
feat: generate constants for ros messages in typescrpit decleration

### PR DESCRIPTION
**Pull request by 27Apoorva**

### CODE REVIEW

File: src/gennodejs/generate.py [hunk 1]
```diff
@@ -818,6 +830,7 @@ def write_msg_types(s, spec):
 
         for field in fields:
             s.write('public {}: {};'.format(field.name, get_js_type(field, spec.package)))
+        write_type_constants(s,spec)
```
Comment (ccpowers) [inline src/gennodejs/generate.py:831]: we need this in msgs too, not just services.

File: src/gennodejs/generate.py [hunk 2]
```diff
@@ -728,6 +728,18 @@ def write_constants(s, spec):
         s.write('}')
         s.newline()
 
+def write_type_constants(s, spec):
```
Comment (ccpowers) [inline src/gennodejs/generate.py:729]: all built in types can be constants. see:
https://wiki.ros.org/msg#Constants

i think there is a function here somewhere called is_builtin that has a map of built in types.

**Review by ccpowers — CHANGES_REQUESTED** [review]

File: src/gennodejs/generate.py (hunk 1 above)
Comment (27Apoorva) [reply src/gennodejs/generate.py:831]: this is in msgs. But I will add it in services too

File: src/gennodejs/generate.py (hunk 2 above)
Comment (ccpowers) [reply src/gennodejs/generate.py:729]: nvm, I think was really tired when I reviewed this. I see that the else statement will take care of non-string or bool types.

**Review by ccpowers — APPROVED** [review]

**Review by grieneis — CHANGES_REQUESTED** [review]
Tested building against these messages and ran into a few errors.

**Review by dnikitop — APPROVED** [review]

**Review by grieneis — APPROVED** [review]

**Pull request merged**